examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,468 @@
+    union(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+
+Return the union of `C` and `D` (with multiplicity).
+"""
+function Base.union(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+  F = C.eq*D.eq
+  return ProjPlaneCurve(F)
+end
+
+################################################################################
+# Components of the curve
+
+@doc Markdown.doc"""
+    curve_components(C::ProjPlaneCurve)
+
+Return a dictionary containing the irreducible components of `C` and their multiplicity.
+"""
+function curve_components(C::ProjPlaneCurve)
+  if isempty(C.components)
+     D = factor(C.eq)
+     C.components = Dict(ProjPlaneCurve(x) => D.fac[x] for x in keys(D.fac))
+  end
+  return C.components
+end
+
+################################################################################
+# Check irreducibility.
+# TODO: change for a direct irreducibility check when available.
+
+@doc Markdown.doc"""
+    isirreducible(C::ProjPlaneCurve)
+
+Return `true` if `C` is irreducible, and `false` otherwise.
+"""
+function Oscar.isirreducible(C::ProjPlaneCurve)
+  if isempty(C.components)
+     D = factor(C.eq)
+     C.components = Dict(ProjPlaneCurve(x) => D.fac[x] for x in keys(D.fac))
+  end
+  if length(C.components) != 1
+     return false
+  else
+     return all(isone, values(C.components))
+  end
+end
+
+################################################################################
+# Check reducedness by computing a factorization
+# TODO: change for a direct squarefree check when available.
+
+@doc Markdown.doc"""
+    isreduced(C::ProjPlaneCurve)
+
+Return `true` if `C` is reduced, and `false` otherwise.
+"""
+function Oscar.isreduced(C::ProjPlaneCurve)
+  if isempty(C.components)
+     D = factor(C.eq)
+     C.components = Dict(ProjPlaneCurve(x) => D.fac[x] for x in keys(D.fac))
+  end
+  return all(isone, values(C.components))
+end
+
+################################################################################
+# Compute squarefree equation.
+# TODO: change for a direct squarefree computation when available.
+
+@doc Markdown.doc"""
+    reduction(C::ProjPlaneCurve)
+
+Return the projective plane curve defined by the squarefree part of the equation of `C`.
+"""
+function reduction(C::ProjPlaneCurve)
+  if isempty(C.components)
+     f = factor(C.eq)
+     C.components = Dict(ProjPlaneCurve(x) => f.fac[x] for x in keys(f.fac))
+  end
+  F = prod(D -> D.eq, keys(C.components))
+  rC = ProjPlaneCurve(F)
+  rC.components = Dict(ProjPlaneCurve(D.eq) => 1 for D in keys(C.components))
+  return rC
+end
+
+################################################################################
+# gives the common components of two projective plane curves
+
+@doc Markdown.doc"""
+    common_component(C::ProjPlaneCurve{S}, D::ProjPlaneCurve{S}) where S <: FieldElem
+
+Return the projective plane curve consisting of the common component of `C` and `D`, or an empty vector if they do not have a common component.
```

Comment:
I do not expect the following remark to be addressed in this PR, but perhaps at some point in the future.

Note that this behavior violates type stability, which is important for performance. IMHO it would be better if the return value of this function always had the same type. So there are at least two options:

1. either return an empty vector, or a vector with one entry, the common component (and perhaps it would then make sense to rename it to `common_components`, so the name can be reused by other geometric objects in the future where there might be more than one common component? I don't know if that even makes sense, though 🤷 )
2. throw an error if there is no common component, and then add a function to check whether there is a common component in the first place (if there isn't such a function already)

Even if one doesn't care about type stability, arguably it'd make more sense to return the value `nothing` instead of an empty vector.